Sources/TuistCore/Utils/XCFrameworkMetadataProvider.swift

```diff
@@ -0,0 +1,26 @@
+import Basic
+import Foundation
+import TuistSupport
+
+public protocol XCFrameworkMetadataProviding {
+    func libraries(
+        frameworkPath: AbsolutePath,
+        fileHandler: FileHandling
+    ) throws -> [XCFrameworkInfoPlist.Library]
+}
+
+public class XCFrameworkMetadataProvider: XCFrameworkMetadataProviding {
+
+    public func libraries(
+        frameworkPath: AbsolutePath,
+        fileHandler: FileHandling = FileHandler.shared
```

Comment:
No need to do dependency injection here because the shared instance is easily mockable.

Reply:
We can define the method in one line.